### ISSUE TITLE
Back button is not correctly managed

### ISSUE THREAD

**Issue by ppaysant** [owncloud/files_pdfviewer#28]
Hello,

I know that back button is a dev nightmare and should be banned, but, hey, it exists and users are using it :)

### Steps to reproduce
1. Go to file list (/index.php/apps/files/)
2. Navigate a little in subdirectories (just to get a history with owncloud urls)
3. Click on a pdf document to view it with this app
4. Click on back button (you may also use alt+arrows to navigate in the history)

### Expected behaviour
If the url changes (back button), the pdf should not be displayed anymore, as the user most probably want to go back to the files list.

### Actual behaviour
The url in the browser url (in url bar) changes correctly but the pdf is always on screen. If you click on the cross to close the viewer, you are on the page as indicated by the url (the navigation really happens but "behind" the pdf viewer).

### Server configuration
**Operating system**: rh6
**Web server:** apache 2.3
**Database:** mariadb
**PHP version:** 5.4.x
**ownCloud version:** (see ownCloud admin page) 7.0.4RC2
**Updated from an older ownCloud or fresh install:** updated
**List of activated apps:** documents, group_custom, gtu, lotsofgroups, user_account_actions
**The content of config/config.php:**

```php
define('DEBUG',true);
$CONFIG = array (
  'instanceid' => 'popopo',
  'passwordsalt' => 'popopo',
  'secret' => 'popopo',
  'trusted_domains' => 
  array (
    0 => 'owncore.chez.moi',
  ),
  'datadirectory' => '/data/owncloud',
  'overwrite.cli.url' => 'https://owncore.chez.moi',
  'dbtype' => 'mysql',
  'version' => '7.0.4.2',
  'dbname' => 'owncloud7',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'popopo',
  'dbpassword' => 'popopo',
  'installed' => true,
  'theme' => 'mycore',
  'maintenance' => false,
  'hasinternetaccess' => false,
  'has_internet_connection' => false,
  'mail_smtpmode' => 'php',
  'backup_file_before_user_deletion' => true,
  'backup_file_dir' => '/tmp/backup/users',
);
```

**Are you using external storage, if yes which one:** none
**Are you using encryption:** no

### Client configuration
**Browser:** firefox / chromium
**Operating system:** debian / sid

**Comment by mjdemilliano** [comment]
+1 I confirm this issue.

**Comment by DeepDiver1975** [comment]
refs https://github.com/owncloud/files_texteditor/issues/50

@PVince81 as multiple editors/viewers share this issue we might want to invent something within core?

**Comment by PVince81** [comment]
I thought I had a ticket already but cannot find it.
So I raised this one: https://github.com/owncloud/core/issues/13414

**Comment by DeepDiver1975** [comment]
THX

**Comment by MorrisJobke** [comment]
@cmonteroluque To be honest: I don't think this is solvable for 8.1.1 :(

**Comment by cmonteroluque** [comment]
8.2 based on https://github.com/owncloud/files_pdfviewer/issues/28#issuecomment-70243715

**Comment by PVince81** [comment]
Managing the back button might not be possible as it requires a new API (viewer API) and have apps implement that API to make it manageable.

However if some apps are misbehaving right now it might be possible to provide a quick fix for every app, making them auto-close whenever the URL changes.

**Comment by fritzw** [comment]
> However if some apps are misbehaving right now it might be possible to provide a quick fix for every app, making them auto-close whenever the URL changes.

I agree that a quick fix would be desirable. However, closing the viewer "when the URL changes" alone is not enough.

As a user, I'd expect the back button to get me back to where I was *before I clicked the PDF file*. Namely, back to the folder where the PDF file resides. Currently, it takes me to *the folder where I was before that* instead (in addition to not closing the PDF viewer). I'd imagine that this would frustrate users no less than the current behavior.

A possible fix would be to include a fragment like `#view=filename.pdf` to the URL when clicking the PDF file. Then, the back button would remove that fragment and the "auto-close whenever the URL changes" behavior would do what users expect.

**Comment by MorrisJobke** [comment]
Fixed in 8.2.2 and 9.0.0 - https://github.com/owncloud/files_pdfviewer/pull/80
